hamclock: New port

Dave-Allured opened the pull request:
#### Description

HamClock is a kiosk-style application that provides real time space weather, radio propagation models, operating events and other information particularly useful to the radio amateur.

https://clearskyinstitute.com/ham/HamClock/

Closes: https://trac.macports.org/ticket/72625

Thanks to @barracuda156 for setting up the initial Portfile.

###### Type(s)

- [x] enhancement

###### Tested on

macOS 13.7.6
Xcode 15.2
CL tools 15.1.0

Also tested on PowerPC.

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with ~~`sudo port -vst install`~~ `sudo port -s install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

pidloop commented:
Works fine on my iMac running macOS 13.7.6. Many thanks for all involved.

Suggest removing dependency on curl and adding -DNO_UPGRADE compile time switch when move to HamClock V4.20.

barracuda156 commented:
The app launches normally on PowerPC and appears functional.

Dave-Allured commented:
> Suggest removing dependency on curl and adding -DNO_UPGRADE compile time switch when move to HamClock V4.20.

* Now updated to 4.20.
* Added `-DNO_UPGRADE` to `CXXFLAGS`.

Dave-Allured commented:
Correction.  That actual command which I don't seem to remember correctly,  is `sudo portindex`.

pidloop commented:
After loading the new Portfile, running `sudo portindex` then `sudo port install hamclock` hamclock version 4.20 is installed and works fine. I also confirmed `-DNO_UPGRADE` was applied (self-updating is disabled).

This on my iMac running macOS 13.7.6, Xcode 15.2, CL tools version: 15.1.0.0.1.1700200546.

Excellent!

pidloop commented:
Hoping this goes live soon.... thanks everyone.

JDLH commented:
I tested this just now. It builds, installs, runs just fine, apparently.

I observed this warning, during the extraction phase:
```
... [elided] ...
--->  Extracting hamclock
Warning: reinplace s|-march=native|| didn't change anything in /opt/local/var/macports/build/_Users_<myuser>_macports_local_x11_hamclock/hamclock/work/ESPHamClock/Makefile
--->  Configuring hamclock
... [elided] ...
```

My environment is:
```
% sw_vers; xcodebuild -version; pkgutil --pkg-info=com.apple.pkg.CLTools_Executables; /usr/bin/clang --version
ProductName:		macOS
ProductVersion:		14.7.6
BuildVersion:		23H626
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
package-id: com.apple.pkg.CLTools_Executables
version: 16.2.0.0.1.1733547573
volume: /
location: /
install-time: 1734383583
Apple clang version 16.0.0 (clang-1600.0.26.6)
Target: arm64-apple-darwin23.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```
